public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/dataHooks.ts

```diff
@@ -97,7 +97,7 @@ export const useResourceGroups: DataHook = (query, datasource, onChange, setErro
       const results = await datasource.getResourceGroups(subscription);
       const options = results.map(toOption);
 
-      if (resourceGroup && !hasOption(options, resourceGroup)) {
+      if (resourceGroup && !datasource.isTemplateVariable(resourceGroup) && !hasOption(options, resourceGroup)) {
```

Comment:
nit: looks like these checks are similar in other places, could extract to a function or something

Reply:
`isTemplateVariable` was the extracting it out to a separate function  😅

Reply:
yeah i thought about that, but because I need datasource it would end up being `isValidOption(value, options, datasource)` which did seem like it wasnt worth it? idk. thoughts?